consensus/polybft/consensus_metrics.go

```diff
@@ -0,0 +1,53 @@
+package polybft
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/armon/go-metrics"
+)
+
+const (
+	// consensusMetricsPrefix is a consensus-related metrics prefix
+	consensusMetricsPrefix = "consensus"
+)
+
+// updateBlockMetrics updates various metrics based on the given block
+// (such as block interval, number of transactions and block rounds metrics)
+func updateBlockMetrics(block *types.Block, blockchain blockchainBackend) error {
+	if block.Number() > 1 {
+		// retrieve parent header
+		parentHeader, parentExists := blockchain.GetHeaderByNumber(block.Number() - 1)
```

Comment:
I would prefer if we do not have to query the previous block again to report this metric.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1040/commits/c4dc3ea72352684c9cade0eab9398d6bdddbdee3